docs/src/tutorials/temporal-properties.md

```diff
@@ -0,0 +1,416 @@
+    /\ __loop_♢isGreen = FALSE
+    /\ __loop_isGreen = FALSE
+    /\ __loop_requestedGreen = FALSE
+    /\ ☐(requestedGreen ⇒ ♢isGreen) = FALSE
+    /\ ☐(requestedGreen ⇒ ♢isGreen)_unroll = TRUE
+    /\ requestedGreen ⇒ ♢isGreen = TRUE
+    /\ ♢isGreen = FALSE
+    /\ ♢isGreen_unroll = FALSE
+    /\ isGreen = FALSE
+    /\ requestedGreen = FALSE *)
+State0 ==
+  RequestWillBeFulfilled_init = FALSE
+    /\ __loop_InLoop = FALSE
+    /\ __loop___temporal_t_1 = FALSE
+    /\ __loop___temporal_t_2 = TRUE
+    /\ __loop___temporal_t_3 = FALSE
+    /\ __loop_isGreen = FALSE
+    /\ __loop_requestedGreen = FALSE
+    /\ __temporal_t_1 = FALSE
+    /\ __temporal_t_1_unroll = TRUE
+    /\ __temporal_t_2 = TRUE
+    /\ __temporal_t_3 = FALSE
+    /\ __temporal_t_3_unroll = FALSE
+    /\ isGreen = FALSE
+    /\ requestedGreen = FALSE
+```
+
+Two things are notable:
+1. The initial state formula appears twice, once as a comment and once in TLA+.
+2. There are way more variables than the two variables we specified.
+
+The comment and the TLA+ specification express the same state, but in the comment, some variable names from the encoding have been replaced
+with more human-readable names.
+For example, there is a variable called `☐(requestedGreen ⇒ ♢isGreen)` in the comment,
+which is called `__temporal_t_1` in TLA+.
+In the following, let's focus on the content of the comment, since it's easier to understand what's going on.
+
+There are many additional variables in the counterexample because to check temporal formulas, Apalache uses an
+encoding that transforms temporal properties to invariants.
+If you are interested in the technical details, the encoding is described in sections 3.2 and 4 of [Biere et al.][].
+However, to understand the counterexample, you don't need to go into the technical details of the encoding.
+We'll go explain the counterexample in the following.
+
+## Counterexamples encode traces with a loop
+
+First, it's important to know that counterexamples to temporal properties are, in general, traces ending in a loop.
+A loop is a partial trace that starts and ends with the same state. 
+A trace containing a loop describes a possible infinite execution that repeats the loop forever.
+
+For example, what is a trace that is a counterexample to the property `♢isGreen`?
+It's an execution that loops without ever finding a state that satisfies `isGreen`.
+For example, a counterexample trace might visually look like this:
+
+![A counterexample trace for the property <>isGreen](img/looping_trace.png)
+
+In contrast, as long as the model checking engine has not found a loop, there may still exist some future state satisfying `isGreen`.
+
+## Utilizing auxiliary variables to find loops
+
+The encoding for temporal properties involves lots of auxiliary variables.
+While some can be very helpful to understand counterexamples,
+many are mostly noise.
+
+Let's first understand how Apalache can identify looping executions using auxiliary variables.
```

Comment:
Let's use the word "lasso" for executions with a loop, to have the stable vocabulary.